cli/command_repository_upgrade.go

```diff
@@ -72,9 +72,19 @@ func (c *commandRepositoryUpgrade) setup(svc advancedAppServices, parent command
 
 	rollbackCmd.Action(svc.directRepositoryWriteAction(c.forceRollbackAction))
 
+	validateCmd := parent.Command("validate", "Validate the upgraded indexes.")
+	validateCmd.Flag("force", "Force rollback the repository upgrade, this action can cause repository corruption").BoolVar(&c.forceRollback)
+
+	validateCmd.Action(svc.directRepositoryWriteAction(c.validateAction))
+
 	c.svc = svc
 }
 
+func (c *commandRepositoryUpgrade) validateAction(ctx context.Context, rep repo.DirectRepositoryWriter) error {
+	_, err := rep.ContentManager().SharedManager.ValidateIndexes(ctx)
+	return err
```

Comment:
Should not we check if the returned report length is more than zero and fail the upgrade if it is ?

Reply:
Noticed that we have not yet hooked this up in a way that will prevent `commitUpgrade()`. I am guessing you want to do that in a separate PR ?

Reply:
@Shrekster yes.  This is code to show how an index comparison can be done.  Reporting the results is viewed as low-risk and so was not included in this code (proof of concept of index comparison).